File: problems/LeetCode/double_pointers/twoSum.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1

        while(True):
            current = numbers[left] + numbers[right]
            if current == target:
                return [left + 1, right + 1 ]
            if current > target:
                right -= 1
            else:
          			left += 1

File: problems/LeetCode/TopInterview150/twoSum.ts
function twoSum(numbers: number[], target: number): number[] {
  let pl = 0;
  let pr = numbers.length - 1;

  while (pl < pr) {
    if (numbers[pl] + numbers[pr] === target) {
      return [pl + 1, pr + 1];
    }

    if (numbers[pl] + numbers[pr] < target) {
      pl++;
    } else {
      pr--;
    }
  }
  return [];
}

// Time complexity: O(n)
// Space complexity: O(1)
